src/components/magnet/magnet-util.ts

```diff
@@ -1,94 +1,46 @@
-                    (bField2[0] * sinAngleRadsMagnet2 + bField2[1] * cosAngleRadsMagnet2);
-    bFieldSum = [bFieldX, bFieldY];
-  } else {
-    bFieldSum = bField1;
-  }
+const kChargeAmount = 1000;
 
-  const scale = 10E4;
-  const magnitude = Math.sqrt(bFieldSum[0] * bFieldSum[0] + bFieldSum[1] * bFieldSum[1]) * scale;
-  const angle = Math.atan2(bFieldSum[1], bFieldSum[0]);
-  return [magnitude, angle];
+interface PointCharge {
+  x: number;
+  y: number;
+  charge: number;
 }
 
-function getBComputeField(xm: number, ym: number) {
-  const b = [0, 0];
-  let dx;
-  let dy;
-  let r2;
-  let r;
-  let r3;
-  let cos;
-  let sin;
-
-  const numDipoles = 22;
-  const m = 0.05;
+export function getFieldMagnitudeAndDirection(magnets: PossibleMagnet[], x: number, y: number) {
+  const fieldTotal = magnets.reduce((acc: Vector, magnet: PossibleMagnet) => {
+    if (magnet) {
+      return acc.add(getFieldForMagnet(magnet, x, y));
+    } else {
+      return acc;
+    }
+  }, new Vector(0, 0));
 
-  const dipoles: number[][] = [];
-  const dx2 = 1 / (numDipoles - 2);    // set up dipoles under bar magnet
+  return [fieldTotal.length(), fieldTotal.toAngle()];
+}
 
-  for (let i = 0; i < numDipoles; i += 2) {
-    dipoles[i] = [];
-    dipoles[i][0] = -0.5 + i * dx2;   // x position of first line of dipoles bottom
-    dipoles[i][1] = -0.04;            // y position move down by -0.04 of first line of dipoles bottom
+function getFieldForMagnet(magnet: Magnet, x: number, y: number) {
+  const plusField = getFieldForPointCharge({
+    x: magnet.x + (magnet.length) / 2,
+    y: magnet.y,
+    charge: kChargeAmount
+  }, x, y);
+  const minusField = getFieldForPointCharge({
+    x: magnet.x - (magnet.length) / 2,
+    y: magnet.y,
+    charge: -kChargeAmount
+  }, x, y);
+
+  return plusField.add(minusField);
+}
 
-    dipoles[i + 1] = [];
-    dipoles[i + 1][0] = -0.5 + i * dx2; // x position of 2nd line of dipoles top
-    dipoles[i + 1][1] = 0.04;             // y position move up by +0.04 of top line of dipoles bottom
-  }
+function getFieldForPointCharge(pointCharge: PointCharge, x: number, y: number) {
+  const chargeVec = new Vector(pointCharge.x, pointCharge.y);
+  const pointVec = new Vector(x, y);
+  const distanceVec = pointVec.subtract(chargeVec);
+  const distanceUnitVec = distanceVec.unit();
 
-  for (let i = 0; i < numDipoles; i++) {
-    dx = xm - dipoles[i][0]; // (xMagnet*cs+yMagnet*sc);
-    dy = ym - dipoles[i][1]; // (-xMagnet*sc+yMagnet*cs);
-    r2 = dx * dx + dy * dy;
-    r = Math.sqrt(r2);
-    r3 = r2 * r;
-    cos = dx / r;
-    sin = dy / r;
-    // taken from http://en.wikipedia.org/wiki/Magnetic_moment
-    b[0] += m * (3 * cos * cos - 1) / r3;   // x component
-    b[1] += m * 3 * sin * cos / r3;         // y component
-    if (r2 === 0) {
-      b[0] = b[1] = 0;
-    }
-  }
-  return b;
+  return distanceUnitVec.multiply(pointCharge.charge / (distanceVec.length() ** 2));
```

Comment:
This is the crux of the math: assume the force of any point charge at a coordinate is proportional to the charge divided by the distance squared. All that's happening in the end is we're summing those forces over all magnets (and in the next commit, assuming multiple dipoles per magnet)